### PR TITLE
micronaut OperationApiController

### LINKED ISSUE #20260: convert OperationApiController to Micronaut
part of #13274

### PR DESCRIPTION
**Pull request by colesnodgrass**
## What
- fix #20260

## How
- convert `OperationApiController` over to Micronaut
- add nginx settings
- add Micronaut annotations to `OperationApiController`
- remove `OperationApiFactory`
- remove `OperationApiBinder`
- add `@Singleton` and `@Inject` annotations to `OperationsHandler`

### CODE REVIEW

**Comment by jdpgrailsdev** [comment]
@colesnodgrass Looks like there is a failing acceptance test related to this change:

```
        Caused by:
        io.airbyte.api.client.invoker.generated.ApiException: createOperation call failed with: 500 - ***
        "servlet":"org.glassfish.jersey.servlet.ServletContainer-7acb5200",
        "message":"Request failed.",
        "url":"/api/v1/operations/create",
        "status":"500"
        ***
            at app//io.airbyte.api.client.generated.OperationApi.getApiException(OperationApi.java:76)
            at app//io.airbyte.api.client.generated.OperationApi.createOperationWithHttpInfo(OperationApi.java:194)
            at app//io.airbyte.api.client.generated.OperationApi.createOperation(OperationApi.java:172)
            at app//io.airbyte.test.utils.AirbyteAcceptanceTestHarness.createOperation(AirbyteAcceptanceTestHarness.java:534)
            at app//io.airbyte.test.acceptance.AdvancedAcceptanceTests.testManualSync(AdvancedAcceptanceTests.java:144)
```

File: airbyte-server/src/main/java/io/airbyte/server/handlers/OperationsHandler.java [hunk 1]
```diff
@@ -25,28 +25,32 @@
...
 import java.util.UUID;
 import java.util.function.Supplier;
 
+@Singleton
 public class OperationsHandler {
 
   private final ConfigRepository configRepository;
   private final Supplier<UUID> uuidGenerator;
 
+  @Inject
```
Comment (benmoriceau) [inline airbyte-server/src/main/java/io/airbyte/server/handlers/OperationsHandler.java:43]: Is the `@Inject` needed?

**Review by benmoriceau — APPROVED** [review]

File: airbyte-server/src/main/java/io/airbyte/server/handlers/OperationsHandler.java (hunk 1 above)
Comment (colesnodgrass) [reply airbyte-server/src/main/java/io/airbyte/server/handlers/OperationsHandler.java:43]: I added it mostly for documentation purposes as there are two constructors on this class.  Technically [it isn't needed](https://docs.micronaut.io/latest/guide/#beans) as 
> Micronaut supports the following types of dependency injection:
> - Constructor injection (must be one public constructor or a single constructor annotated with @Inject)
and there is only a single public constructor in this class.

**Comment by colesnodgrass** [comment]
@jdpgrailsdev I started looking into that failure late yesterday afternoon, still digging into it this morning.

**Comment by colesnodgrass** [comment]
After talking with Benoit, going to merge this in with the broken acceptance tests as we're running into issues with acceptance tests in general and that way we can fix them all in the single branch (hopefully at once).

**Pull request merged**